addons/stock/models/stock_production_lot.py

```diff
@@ -215,3 +215,9 @@ def _find_delivery_ids_by_lot(self, lot_path=None, delivery_by_lot=None):
                     delivery_ids.add(line.picking_id.id)
             delivery_by_lot[lot.id] = list(delivery_ids)
         return delivery_by_lot
+
+    @api.model
+    def check_lots_exist(self, lots):
+        """returns the list of serial numbers that do not exist"""
+        existing_sn = self.env['stock.production.lot'].search([('name', 'in', lots)])
+        return list(set(lots) - set(existing_sn.mapped('name')))
```

Comment:
Perhaps you can just put this in the `stock_picking.py` file in `point_of_sale` addon. I don't see a reason to ask the inventory team to approve such method.